akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/ClusterSharding.scala

```diff
@@ -273,7 +275,14 @@ final class Entity[M, E] private (
       entityProps,
       settings,
       Optional.ofNullable(newExtractor),
-      allocationStrategy)
+      allocationStrategy,
+      role)
+
+  /**
+   *  Run the Entity actors on nodes with the given role.
+   */
+  def withRole(role: String): Entity[M, E] =
+    copy(role = Optional.ofNullable(role))
```

Comment:
not added to the existing settings if they exist as users could get a surprise if the order the call the with methods changes behavior and if there isn't a settings we need an actor system to create it